src/drivers/dw/dma.c

```diff
@@ -975,10 +975,11 @@ static void dw_dma_verify_transfer(struct dma *dma, unsigned int channel,
 		dw_write(dma, DW_DMA_CHAN_EN, DW_CHAN_MASK(channel));
 		/* fallthrough */
 	default:
-		if (ll_uncached->ctrl_hi & DW_CTLH_DONE(1)) {
+		while (ll_uncached->ctrl_hi & DW_CTLH_DONE(1)) {
 			ll_uncached->ctrl_hi &= ~DW_CTLH_DONE(1);
 			chan->lli_current =
 				(struct dw_lli *)chan->lli_current->llp;
+			ll_uncached = cache_to_uncache(chan->lli_current);
 		}
```

Comment:
@tlauda is there a situation where chan->lli_current can reach to NULL? I mean we are traversing the list. Is there a way we can reach the end of the list?

Reply:
It's not possible, the list is cyclic. Also we are deasserting the DONE bit, so we'll eventually exit the loop. Adding some additional exit criteria is unnecessary.